pages/airdrop/index.tsx

```diff
@@ -1,46 +1,83 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
-import { useEffect, useState } from 'react'
+import { useState } from 'react'
 
+import { useMutation } from 'react-query'
+import axios from 'axios'
+import Web3 from 'web3'
 import { Box, Button, Center, Flex, Input, Text, useToast } from '@chakra-ui/react'
 
 import Layout from '../../src/components/layout'
-import { useSearchPublicEntity, useCreatePublicEntity } from '../../src/hooks/usePublicEntities'
+import Spinner from '../../src/components/Spinner/Spinner'
 
 const Airdrop = () => {
   const toast = useToast()
-  const [searchingAddress, setSearchingAddress] = useState('')
-  const [claimedStatus, setClaimedStatus] = useState('')
+  const [claimantAddress, setClaimantAddress] = useState('')
+  const [claimantEmail, setClaimantEmail] = useState('')
+  const [claimantDiscord, setClaimantDiscord] = useState('')
+
+  const web3 = new Web3()
+
+  const ENTITY_API = process.env.NEXT_PUBLIC_ENTITY_API_URL
+  const WHITELIST_EVENT_COLLECTION_ID = process.env.NEXT_PUBLIC_WHITELIST_EVENT_COLLECTION_ID
+
+  const onSuccess = (data: any) => {
+    toast({
```

Comment:
This is is duplication of https://github.com/bugout-dev/portal/blob/2cf35cf045cd1e22de2fab05c10da925af088278/pages/terminus/index.tsx#L67
Can we have toas component which we will be reuse with different argument `message`?